### ISSUE TITLE
Confused about validation data  in Keras

### ISSUE THREAD

**Issue by Yingyingzhang15** [fchollet/keras#2128]
Hi.I'm confused about the validation data.

My code is:
     model.fit(X_train, y_train, batch_size=batch_size, nb_epoch=nb_epoch,verbose=2, 
                   validation_data=  (X_val, y_val))
1. I understand that the validation data is used to evaluate the accuracy of model after each epoch. It helps when I use earlystopping. Is there any other functions of validation data?
2. The Keras documents says that:
   If you set the validation_split argument in model.fit to e.g. 0.1, then the validation data used will be the last 10% of the data. If you set it to 0.25, it will be the last 25% of the data, etc. 
   If I want to have choose validation data randomly, what should I do?
3. Can I use Cross-validation in Keras? If yes, can you tell me how to use it in Keras?

Thank you very much in advance!

**Comment by EderSantana** [comment]
The FAQ got you figured: http://keras.io/faq/#how-can-i-interrupt-training-when-the-validation-loss-isnt-decreasing-anymore

**Comment by Yingyingzhang15** [comment]
@EderSantana Thank you for your answer! I have read the FAQ before, but I still feel confused.

> 2. The Keras documents says that:
If you set the validation_split argument in model.fit to e.g. 0.1, then the validation data used will be the last 10% of the data. If you set it to 0.25, it will be the last 25% of the data, etc.
This is what the FAQ says.  If I want to have choose validation data randomly, what should I do?

> 3.Can I use Cross-validation in Keras? If yes, can you tell me how to use it in Keras?
Can you answer this question?

 I'm new to Keras, and I want to have a good understanding of it. Thank you very much !

**Comment by rpinsler** [comment]
If you want to choose it randomly, shuffle your training data first, then let keras pick the last p%. Alternatively, you can directly input your own validation data via the `validation_data` argument.

The use of cross-validation is discussed here in #1711

**Comment by Yingyingzhang15** [comment]
@rpinsler Thank you very much! :)
